From e47ef39792238aa487ded23034ea2198b08dcd16 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 15 Feb 2016 19:23:45 +0100
Subject: [PATCH] net: increase IPV6_MAX_ADDRESSES to 64

With IPv6 privacy extensions enabled, ULA prefixes and long running
connections keeping older temporary addresses in use, the previous
limit for IPV6_MAX_ADDRESSES (16) can run out within a day.

This avoids:

IPv6: ipv6_create_tempaddr: regeneration time exceeded - disabled
      temporary address support

which has fatal consequences even for IPv4 functionality.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/net/addrconf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/net/addrconf.h
+++ b/include/net/addrconf.h
@@ -14,7 +14,7 @@
 
 #define ADDR_CHECK_FREQUENCY		(120*HZ)
 
-#define IPV6_MAX_ADDRESSES		16
+#define IPV6_MAX_ADDRESSES		64
 
 #define ADDRCONF_TIMER_FUZZ_MINUS	(HZ > 50 ? HZ / 50 : 1)
 #define ADDRCONF_TIMER_FUZZ		(HZ / 4)
